processor/src/test/java/org/mapstruct/ap/test/nullvaluepropertymapping/NullValuePropertyMappingTest.java

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright MapStruct Authors.
+ *
+ * Licensed under the Apache License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package org.mapstruct.ap.test.nullvaluepropertymapping;
+
+import java.util.Arrays;
+
+import org.junit.Ignore;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mapstruct.ap.testutil.IssueKey;
+import org.mapstruct.ap.testutil.WithClasses;
+import org.mapstruct.ap.testutil.compilation.annotation.CompilationResult;
+import org.mapstruct.ap.testutil.compilation.annotation.Diagnostic;
+import org.mapstruct.ap.testutil.compilation.annotation.ExpectedCompilationOutcome;
+import org.mapstruct.ap.testutil.runner.AnnotationProcessorTestRunner;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * @author Sjaak Derksen
+ */
+@RunWith(AnnotationProcessorTestRunner.class)
+@WithClasses({
+    Address.class,
+    Customer.class,
+    CustomerDTO.class,
+    AddressDTO.class,
+    HomeDTO.class,
+    UserDTO.class
+})
+public class NullValuePropertyMappingTest {
+
+    @Test
+    @WithClasses(CustomerMapper.class)
+    @IssueKey("1306")
```

Comment:
Isn't it the entire class for 1306?

Reply:
I got the reproducer form #1306. But in essence the answer is a bit larger than the original issue covered. So I think it deserved a package of its own. I can stick it to the class.